ambry-clustermap/src/main/java/com/github/ambry/clustermap/DynamicClusterChangeHandler.java

```diff
@@ -310,81 +303,79 @@ private void addOrUpdateInstanceInfos(List<InstanceConfig> instanceConfigs) thro
   /**
    * Update info of an existing instance. This may happen in following cases: (1) new replica is added; (2) old replica
    * is removed; (3) replica's state has changed (i.e. becomes seal/unseal).
-   * @param instanceConfig the {@link InstanceConfig} used to update info of instance.
+   * @param dataNodeConfig the {@link DataNodeConfig} used to update info of instance.
    * @return a pair of lists: (1) new added replicas; (2) removed old replicas, during this update.
    */
-  private Pair<List<ReplicaId>, List<ReplicaId>> updateInstanceInfo(InstanceConfig instanceConfig) throws Exception {
+  private Pair<List<ReplicaId>, List<ReplicaId>> updateInstanceInfo(DataNodeConfig dataNodeConfig) throws Exception {
     final List<ReplicaId> addedReplicas = new ArrayList<>();
     final List<ReplicaId> removedReplicas = new ArrayList<>();
-    String instanceName = instanceConfig.getInstanceName();
+    String instanceName = dataNodeConfig.getInstanceName();
     logger.info("Updating replicas info for existing node {}", instanceName);
-    List<String> sealedReplicas = getSealedReplicas(instanceConfig);
-    List<String> stoppedReplicas = getStoppedReplicas(instanceConfig);
+    Set<String> sealedReplicas = dataNodeConfig.getStoppedReplicas();
+    Set<String> stoppedReplicas = dataNodeConfig.getSealedReplicas();
```

Comment:
Sorry, I didn't catch this when I was reviewing the code. It mistakenly assign `StoppedReplicas` to `sealedReplicas`